javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -812,7 +826,7 @@ public T head() {
         if (isEmpty()) {
             throw new NoSuchElementException("head of empty Vector");
         } else {
-            return get(0);
+            return leading[leadingOffset];
```

Comment:
`head` will always be in the `leading`, no need to run the checks present in `get` (it's a lot faster this way)